bin/serve_update_description_gen/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/GrigoryKrasnochub/updaterini"
+	"github.com/urfave/cli/v2"
+)
+
+const descriptionNameSeparator = "\\\\//"
+const descriptionFilename = "description.txt"
+const outputFilename = "serv_update.json"
+
+func main() {
+	app := &cli.App{
+		Name:  "serve update description generator",
+		Usage: "generate json description file for serve update source of updaterini library",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "url",
+				Aliases:  []string{"u"},
+				Required: true,
+				Usage:    "base url for all versions, (base url + version folder + asset file name) is used by library to load asset from server",
+			},
+			&cli.StringFlag{
+				Name:    "descFilename",
+				Aliases: []string{"d"},
+				Value:   descriptionFilename,
+				Usage: "filename in version folder, that contains version description, " +
+					"will be parsed and skipped in assets. Parsed description will be used in result json file",
+			},
+			&cli.StringFlag{
+				Name:    "descNameSeparator",
+				Aliases: []string{"s"},
+				Value:   descriptionNameSeparator,
+				Usage:   "separator that separate version name from version description in version description file",
+			},
+			&cli.PathFlag{
+				Name:     "inputDir",
+				Aliases:  []string{"id"},
+				Required: true,
+				Usage:    "path to versions folder",
+			},
+			&cli.PathFlag{
+				Name:    "outputFilepath",
+				Aliases: []string{"of"},
+				Usage:   "path to place generator output file",
+			},
+		},
+		Action: func(context *cli.Context) error {
+			var err error
+			outputFilepath := context.Path("outputFilepath")
+			if outputFilepath == "" {
+				outputFilepath, err = os.Executable()
+				if err != nil {
+					return err
+				}
+				outputFilepath = filepath.Join(filepath.Dir(outputFilepath), outputFilename)
+			}
+			vReader := verReader{
+				versionsDir:              context.Path("inputDir"),
+				descriptionFilename:      context.String("descFilename"),
+				descriptionNameSeparator: context.String("descNameSeparator"),
+			}
+			versions, err := vReader.readVersionsDir()
+			if err != nil {
+				return err
+			}
+			baseUrl := context.String("url")
+			if !strings.HasSuffix(baseUrl, "/") {
+				baseUrl += "/"
+			}
+			for i, ver := range versions {
+				versions[i].VersionFolderUrl = baseUrl + ver.Version + "/"
+			}
+			jsonVersions, err := json.Marshal(versions)
```

Comment:
Generated json field names is different from example file. Need to either fix field names with `json` tags or fix the example file

Generated:
```json
[
  {
    "folder_url": "build.example.com/v0.5.0/",
    "Name": "version name",
    "Description": "description of the version",
    "Version": "v0.5.0",
    "Assets": [
      {
        "Filename": "awl-android-v0.5.0.apk"
      }
    ]
  }
]
```

From example:
https://github.com/GrigoryKrasnochub/updaterini/blob/65ab3cbb26d40f3881f4f016e91bd8829aa3af8e/bin/serve_update_description_gen/example.json#L1-L16

Reply:
fixed!